engine/src/main/java/org/terasology/rendering/dag/nodes/PrePostCompositeNode.java

```diff
@@ -150,6 +148,8 @@ public PrePostCompositeNode(Context context) {
         if (hazeIsEnabled) {
             addDesiredStateChange(setHazeInputTexture);
         }
+
+        addDesiredStateChange(new SwapGBuffers(gBufferPair));
```

Comment:
I'm not sure which commit you're looking at.

In the read/write terminology, I correctly swapped it before the rendering.
In the latest/stale terminology, I correctly swapped it after the rendering.

The latest commit should have the swap -after- the rendering.